src/data/data1d/Data1DManager.ts

```diff
@@ -41,11 +41,11 @@ export function fromCSD(result, options = {}, usedColors = {}) {
   let offset = dimension.coordinatesOffset.magnitude;
 
   let buffer = dependentVariables[0].components[0];
-  let re: number[] = [];
-  let im: number[] = [];
-  for (let i = buffer.length - 1; i > 0; i -= 2) {
-    re.push(buffer[i - 1]);
-    im.push(buffer[i]);
+  const re = new Float64Array(n);
```

Comment:
This code change nothing right ? Just optimization ?

Could you also change the x variable so that it is a Float64Array ?

Could you also avoid the .map for data.im. We have a utility to multiply a xArray

Could you add a comment to this function. What is CSD ? (should this method be renamed ?)

Reply:
Ok I understood. It is the length of the array that is based on dimension.count and not on the buffer anymore.

Reply:
I feel like this code should be somewhere in the parser, not in nmrium

Reply:
Yes I agree. @jobo322 why is this code not in nmr-parser ?

We have a lot of changes in the data structure in NMRium while the goal on nmr-parser is to avoid this seems to me.

https://github.com/cheminfo/nmrium/blob/94e9d90fdb1b0ddf90ff25b3755ce35bf7682682/src/data/SpectraManager.ts#L95-L111

In nmr-load-save do we still plan to use nmr-parser ?

Reply:
nmr-load-save will export the nmrium format

Reply:
> Ok I understood. It is the length of the array that is based on dimension.count and not on the buffer anymore.

@lpatiny yes, dimension.count could be bigger than buffer length (small difference), so the idea is to fill with zeros. 

Is strange that `count` and data length do not match, julien did use `count` and he was working with a guy that understands JEOL data.